src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/StoredProcedures/IStoredProcedure.cs

```diff
@@ -2,15 +2,14 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
 // -------------------------------------------------------------------------------------------------
-
+using System.Threading;
+using System.Threading.Tasks;
 using Microsoft.Azure.Cosmos.Scripts;
 
 namespace Microsoft.Health.Fhir.CosmosDb.Features.Storage.StoredProcedures
 {
     public interface IStoredProcedure
     {
-        string FullName { get; }
-
-        StoredProcedureProperties ToStoredProcedureProperties();
+         Task<StoredProcedureExecuteResponse<T>> ExecuteStoredProc<T>(Scripts client, string partitionId, CancellationToken cancellationToken, params object[] parameters);
```

Comment:
Probably it makes sense to rename this method to "ExecuteStoredProc**Async**"

Reply:
done